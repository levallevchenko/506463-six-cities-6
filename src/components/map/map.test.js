import React from 'react';
import {render} from '@testing-library/react';
// import {cardsFull} from '../../tests-mock';
import Map from './map';

describe(`Map component test`, () => {
  const location = {};
  const points = [{location: {latitude: 1, longitude: 1}}];
  const currentOffer = {location: {latitude: 1, longitude: 1}};

  it(`Main Map component should render correctly`, () => {
    const {container} = render(
        <Map
          location={location}
          points={points}
          isMainMap={true}
          currentOffer={currentOffer} />
    );

    expect(container).toMatchSnapshot();
  });

  it(`Room Map component should render correctly`, () => {
    const {container} = render(
        <Map
          location={location}
          points={points}
          isMainMap={false}
          currentOffer={currentOffer} />
    );

    expect(container).toMatchSnapshot();
  });
});
