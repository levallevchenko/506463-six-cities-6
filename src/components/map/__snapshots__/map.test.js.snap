// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component test Main Map component should render correctly 1`] = `
<div>
  <section
    class="map cities__map leaflet-container leaflet-grab leaflet-touch-drag"
    id="map"
    style="height: auto; position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      />
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      />
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      >
        <div
          class="leaflet-control-zoom leaflet-bar leaflet-control"
        >
          <a
            aria-label="Zoom in"
            class="leaflet-control-zoom-in"
            href="#"
            role="button"
            title="Zoom in"
          >
            +
          </a>
          <a
            aria-label="Zoom out"
            class="leaflet-control-zoom-out"
            href="#"
            role="button"
            title="Zoom out"
          >
            −
          </a>
        </div>
      </div>
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JS library for interactive maps"
          >
            Leaflet
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`Map component test Room Map component should render correctly 1`] = `
<div>
  <section
    class="map property__map leaflet-container leaflet-grab leaflet-touch-drag"
    id="map"
    style="height: 579px; position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      />
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      />
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      >
        <div
          class="leaflet-control-zoom leaflet-bar leaflet-control"
        >
          <a
            aria-label="Zoom in"
            class="leaflet-control-zoom-in"
            href="#"
            role="button"
            title="Zoom in"
          >
            +
          </a>
          <a
            aria-label="Zoom out"
            class="leaflet-control-zoom-out"
            href="#"
            role="button"
            title="Zoom out"
          >
            −
          </a>
        </div>
      </div>
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JS library for interactive maps"
          >
            Leaflet
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;
